src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java

```diff
@@ -414,11 +415,11 @@ private interface Rejector {
         Rejector rejector = null;
         Field field = GroovyCallSiteSelector.field(receiver, attribute);
         if (field != null) {
-            if (whitelist.permitsFieldSet(field, receiver, value)) {
+            if (permitsFieldSet(whitelist, field, receiver, value)) {
                 Checker.preCheckedCast(field.getType(), value, false, false, false);
                 return super.onSetAttribute(invoker, receiver, attribute, value);
             } else {
-                throw StaticWhitelist.rejectField(field);
+                rejector = () -> rejectField(field);
```

Comment:
Intentionally changing this to delayed?

Reply:
Ah yeah, I forgot to note that this was also an oversight in https://github.com/jenkinsci/script-security-plugin/commit/85d16bd851b355458f6396dbc4f92c3e0eebcd72. I think the effect is that the attempt to make static fields accessible via attribute syntax (`.@`) in that commit did not work correctly when modifying the attribute, only when getting it. Before that commit it did not work in either case, and after this commit it will work in both cases.

Reply:
Actually I think the problems were more limited. If you enter this method with code like `MyClass.staticField = x`, then `GroovyCallSiteSelector.field` could only find fields declared in `java.lang.Class` itself or its supertypes, so `field != null` could only be true if your static field has the same name as a field declared in `java.lang.Class`. In other cases, the code will fall through to the new case from https://github.com/jenkinsci/script-security-plugin/commit/85d16bd851b355458f6396dbc4f92c3e0eebcd72 (which has test coverage), and will work correctly. So for example, if you have a static field named `name`, `MyClass.@name = x` will fail because `field java.lang.Class name` is not allowed to be used. It didn't work correctly before https://github.com/jenkinsci/script-security-plugin/commit/85d16bd851b355458f6396dbc4f92c3e0eebcd72 either though, so impact is minimal.

The other difference would be that if you had code like `object.staticField` for some field that you had approved as `staticField Foo foo` previously, the old code would check if `field Foo foo` was approved, and if not, it would reject it as `field ...`, rather than checking if `staticField ...` is allowed. That also would have been broken before https://github.com/jenkinsci/script-security-plugin/commit/85d16bd851b355458f6396dbc4f92c3e0eebcd72 though.